crates/executor/src/main.rs

```diff
@@ -62,89 +47,38 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
-                        if message.topic ==  gossipsub_ident_topic(Network::Sepolia, Topic::NewJob).into() {
-                            let job: Job = serde_json::from_slice(&message.data)?;
-                            info!("Received a new job event: {}", hash!(&job));
-                            job_record.register_job(job);
+    Executor::new(swarm_events_rx, finished_job_topic_tx, picked_job_topic_tx, runner, prover);
 
-                        }
-                        // Received a picked-job message from the network
-                        if message.topic ==  gossipsub_ident_topic(Network::Sepolia, Topic::PickedJob).into() {
-                            let job: Job = serde_json::from_slice(&message.data)?;
-                            info!("Received picked job event: {}", hash!(&job));
-                            job_record.remove_job(&job);
-                        }
-                    },
-                    Event::Subscribed { peer_id, topic } => {
-                        info!("{} subscribed to the topic {}", peer_id.to_string(), topic.to_string());
-                    },
-                    Event::Unsubscribed { peer_id, topic }=> {
-                        info!("{} unsubscribed to the topic {}", peer_id.to_string(), topic.to_string());
-                    },
-                    _ => {}
-                }
-            },
-            Some(Ok(event_vec)) = event_stream.next() => {
-                debug!("{:?}", event_vec);
-            },
-            Some(Ok(job_trace)) = runner_scheduler.next() => {
-                info!("Scheduled proving of job_trace: {}", hash!(&job_trace));
-                prover_scheduler.push(prover.run(job_trace)?);
-            },
-            Some(Ok(job_witness)) = prover_scheduler.next() => {
-                info!("Calculated job_witness: {}", hash!(&job_witness));
-            },
-            _ = shutdown_signal() => {
-                break
-            }
-            else => break
-        };
+    let server = ExecutorGRPCServer::default();
 
-        if runner_scheduler.len() + prover_scheduler.len() < MAX_PARALLEL_JOBS
-            && !job_record.is_empty()
-        {
-            if let Some(job) = job_record.take_job().await {
-                let serialized_job = serde_json::to_string(&job)?;
-                send_topic_tx.send(serialized_job.into()).await?;
-                info!("Sent picked job event: {}", hash!(&job));
+    let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
+    health_reporter.set_serving::<ExecutorServiceServer<ExecutorGRPCServer>>().await;
 
-                info!("Scheduled run of job: {}", hash!(&job));
-                runner_scheduler.push(runner.run(job)?);
-            }
-        }
-    }
+    Server::builder()
+        .add_service(health_service)
+        .add_service(ExecutorServiceServer::new(server))
+        .serve_with_shutdown("0.0.0.0:50052".parse().unwrap(), shutdown_signal())
```

Comment:
same reason above on delegator
```suggestion
        .serve_with_shutdown("[::]:50052".parse().unwrap(), shutdown_signal())
```